File: 백준/Bronze/1032. 명령 프롬프트/명령 프롬프트.cc
#include <iostream>
#include <vector>
#include <string>

int main() {
  int N;
  std::cin >> N;

  std::vector<std::string> L(N);
  for (int i = 0; i < N; ++i) {
    std::cin >> L[i];
  }

  std::string valid = L[0];
  std::string new_valid;

  for (int i=1;i<N;++i) {
    new_valid="";
    for (int k = 0; k < valid.size(); ++k) {
      if (valid[k] == L[i][k]) {
        new_valid += valid[k];
      }
      else {
        new_valid += "?";
      }
    }
    valid = new_valid;
  }

  std::cout << valid << std::endl;
  return 0;
}

File: 백준/Bronze/1032. 명령 프롬프트/명령 프롬프트.py

N = int(input())
L = []

for _ in range(N):
  text = input()
  L.append(text)
  
valid = L[0]
new_valid = ''
for i in range(1,N):
  x = L[i]
  k=0
  for char in x:
    if char == valid[k]:
      new_valid += char
    else:
      new_valid += '?'
    k += 1
  valid = new_valid
  new_valid = ''
print(valid)